views/start.dt

```diff
@@ -0,0 +1,85 @@
+extends layout
+
+block title
+	- import dubregistry.viewutils;
+	- import dubregistry.web;
+	- import vibe.data.json;
+	- import std.algorithm.iteration : map;
+	- auto title = "Find, Use and Share DUB Packages";
+	script(type="application/javascript", src="scripts/home.js")
+	script(type="application/javascript").
+		window.categories = #{serializeToJson(info.categories).toString()};
+
+block body
+	- auto mirror = req.params.get("mirror", "");
+	- if (mirror.length)
+		p This is a mirror of #[a(href=mirror)= mirror].
+	- else
+		p Welcome to DUB, the D package registry.
+
+	- string makeHomeURL(string sort) {
+	-   return "?sort=" ~ sort;
+	- }
+
+	p Total #{info.packageCount} packages found.
+
+		form#category-form(method="GET", action="")
+			p Select category:
+				select#category(name="category", size="1", onChange='document.getElementById("category-form").submit()')
+					- void outputCat(Category cat)
+						- if (!cat)
+							option(value="") All packages
+						- else
+							option(value=cat.name)= cat.indentedDescription
+						- if (!cat || cat.subCategories.length)
+							- foreach (c; cat ? cat.subCategories : info.categories)
+								- outputCat(c);
+					- outputCat(null);
+				button#category-submit(type="submit") Update
+		#category-dynamic-form(style="display: none")
+			p Select category:
+				- foreach (i; 0 .. 6)
+					select(id="categories_#{i}", name="categories_#{i}", onChange='setCategoryFromSelector(#{i})')
+			:javascript
+				setupCategoryForm();
+
+		- void listPackages(P)(P packages) {
+			ul
```

Comment:
make this ul contain a classname and make the css apply to that `ul.packageList` or similar instead of hardcoding it just for the homepage so it can be reused in other places.